docs/pallas/tpu/collectives.rst

```diff
@@ -0,0 +1,452 @@
+Pallas Collectives
+===================
+Pallas has support for manual DMA (between memories of the same chip) and remote DMA (between memories of different chips) support. It also has an API for nesting pipelines of compute overlapped collectives.
+
+Manual DMAs are a necessary imperative escape hatch for maximum performance, but we hope that nested pipelines will minimize where they are used in codebases.
+
+Manual DMAs
+------------
+
+.. code-block:: python
+
+  from jax.experimental import pallas as pl
+  from jax.experimental.pallas import tpu as pltpu
+
+  def move_data(src_ref, dst_ref, sem, send_sem, recv_sem):
+    local_copy = pltpu.async_copy(src_ref, dst_ref.at[0:8], sem)
+    local_copy.wait()
+
+    # Should do a barrier the first time we communicate
+    # with another chip.
+    barrier_sem = pltpu.get_barrier_semaphore()
+    pltpu.semaphore_signal(barrier_sem, device_id=neighbor)
+    pltpu.semaphore_wait(barrier_sem)
+
+    dev_id = pltpu.device_id()
+    other_dev_id = 1 - dev_id
+    remote_copy = pltpu.async_remote_copy(
+      src_ref, dst_ref.at[0:8], send_sem, recv_sem, other_dev_id,
+      device_id_type=pltpu.DeviceIdType.LOGICAL,
+    )
+    # remote_copy.wait() for both.
+    remote_copy.wait_send()
+    remote_copy.wait_recv()
+
+    # Both APIs have make_ variants for building a descriptor without starting
+    # it. This allows us to start a DMA in one grid iteration and wait for it
+    # in another.
+    local_copy = pltpu.make_async_copy(src_ref, dst_ref.at[0:8], sem)
+    local_copy.start()
+    local_copy.wait()
+  
+  src_data = pltpu.VMEM((8,128), jnp.float32)
+  dst_data = pltpu.VMEM((16,128), jnp.float32)
+  semaphore = pltpu.SemaphoreType.DMA
+
+  # Inside a kernel:
```

Comment:
Are both this code and the call below runnable in the same context? Make sure the example runs, as stated.